### PR TITLE
Feature/examples utils - filter

### PR DESCRIPTION
**Pull request by lievenvg**
## PR Checklist

This PR fulfills the following requirements:

- [x] The commit message follows our guidelines: [Contributing guidelines](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] A [changelog entry](https://github.com/digipolisantwerp/acpaas-ui_angular/blob/master/guidelines/CHANGELOG.md) was added

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+			}
+		});
+	});
+	return result;
+};
+
+public changeCheckFilter(value) {
+	// Update filter value
+	this.checkFilter.value = value;
+	// filter data
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Please be consistent if you decide to put comments in examples.

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	this.searchResults = this.filterService.filterData(this.heroList, [this.searchFilter]);
+}`;
+
+public codeExampleJS5 = `this.selectFilter.id = 'selectFilter';
+this.selectFilter.name = 'Select your hero';
+this.selectFilter.options = this.heroList;
+this.selectFilter.value = [];
+this.selectFilter.parse = (data, value) => {
+	if (!value || value.length === 0) {
+			return;
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Fix indenting please.

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	// filter data
+	this.checkResults = this.filterService.filterData(this.heroList, [this.checkFilter]);
+}`;
+
+public codeExampleJS4 = `this.searchFilter.id = 'searchFilter';
+this.searchFilter.name = 'Search here...';
+this.searchFilter.value = '';
+this.searchFilter.parse = (data, value) => {
+	if (!value || value.length === 0) {
+				return ;
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Fix indenting please.

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+}`;
+
+	public codeExampleHTML1 = `<aui-checkbox-filter [filter]="checkFilter" (update)="changeCheckFilter($event)"></aui-checkbox-filter>`;
+
+	public codeExampleHTML2 = `<aui-input-filter [filter]="searchFilter" (update)="changeSearchFilter($event)"></aui-input-filter>`;
+
+	public codeExampleHTML3 = `<aui-select-filter [filter]="selectFilter" (update)="changeSelectFilter($event)"></aui-select-filter>`;
+
+	constructor(public filterService: FilterService) {}
+	ngOnInit() {
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Fix spacing please.

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+		// Checkbox filter
+		this.checkFilter.id = 'checkFilter';
+		this.checkFilter.name = 'Checkbox filter';
+		this.checkFilter.options = this.heroList;
+		this.checkFilter.value = [];
+		this.checkFilter.parse = (data, value) => {
+			if (!value || value.length === 0) {
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Fix indenting please.

File: packages/utils/examples/utils/pages/filter/filter.page.ts [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+			});
+			return result;
+		};
+
+		// Input filter
+		this.searchFilter.id = 'searchFilter';
+		this.searchFilter.name = 'Search here...';
+		this.searchFilter.value = '';
+		this.searchFilter.parse = (data, value) => {
+			if (!value || value.length === 0) {
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/pages/filter/filter.page.ts]: Fix indenting please.

File: packages/utils/examples/utils/utils.module.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterModule } from '@angular/router';
+import {
+	FilterModule,
+	LabelsModule,
+	WindowModule,
```
Comment (TriangleJuice) [inline packages/utils/examples/utils/utils.module.ts]: Suggestion: if you already import these modules, maybe you should add the documentation too?

File: packages/utils/src/lib/filter/README.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        { id: 'id4', name: 'Iron Man' },
+        { id: 'id5', name: 'Deadpool' },
+    ];
+```
+
+```typescript
+constructor(public filterService: FilterService) {}
+```
+
+**Checkbox filter example**
```
Comment (TriangleJuice) [inline packages/utils/src/lib/filter/README.md]: Maybe you should create a heading "Examples" with sub-headings, so it's consistent with the other READMEs.

File: packages/utils/examples/utils/utils.module.ts (hunk 7 above)
Comment (lievenvg) [reply packages/utils/examples/utils/utils.module.ts]: I removed them again, so they can be separate PR's.

File: packages/utils/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# @acpaas-ui/ngx-components/utils
+
+The utils module is a collection of some frequently used utility modules so you don't need to reinvent the wheel every time.
```
Comment (TriangleJuice) [inline packages/utils/README.md]: Actually, I'd prefer to know which utils, when I'm reading this 😉

File: packages/utils/src/lib/window/README.md [hunk 10]
```diff
@@ -12,34 +13,27 @@ import { WindowModule } from '@acpaas-ui/ngx-components/utils'`;
 
 Visit our [documentation site](https://acpaas-ui.digipolis.be/) for full how-to docs and guidelines
 
-### API
+### WINDOW_PROVIDERS
```
Comment (TriangleJuice) [inline packages/utils/src/lib/window/README.md]: Why is this relevant to document?

File: packages/utils/src/lib/filter/README.md [hunk 11]
```diff
@@ -144,7 +144,7 @@ public changeSearchFilter(value) {
 <aui-input-filter [filter]="searchFilter" (update)="changeSearchFilter($event)"></aui-input-filter>
 ```
 
-**Select filter example**
+#### Select filter example
```
Comment (TriangleJuice) [inline packages/utils/src/lib/filter/README.md]: While you're fixing the other remarks, please get rid of all the words 'example' in the titles 😉
